src/Extension/ExtensionManager.php

```diff
@@ -109,16 +99,17 @@ public function getExtensions()
                 $extension->setInstalled(true);
                 $extension->setVersion(Arr::get($package, 'version'));
 
-                $extensions->put($extension->getId(), $extension);
+                // We use the composer package name as the key so that:
+                //   1. We don't have naming collisions
+                //   2. We can use it for constant time lookups in `calculateDependencies`
+                $extensions->put(Arr::get($package, 'name'), $extension);
```

Comment:
I am not entirely sure this causes issues with extensions using the Extension Manager, @datitisev and @clarkwinkelmann do you know of any extensions that will be impacted by this?

I'm okay with this solution, but we do need to clearly mention this in the upgrade notes for devs.